onnxruntime/core/session/inference_session.h

```diff
@@ -884,6 +888,7 @@ class InferenceSession {
     }
 
     IExecutionProvider* cached_execution_provider_for_graph_replay_ = nullptr;
+    int graph_annotation_id_ = 0;
```

Comment:
Please do not store run option in session object. 
Consider the case that a user have multiple threads to call session run, and two parallel runs with different cuda graph ID.

Reply:
It will be interesting to see if we can allow multi-threaded CUDAGraph usage now with this feature. Per CUDA documentation, using the same CUDAGraph instance from multiple threads is not supported (https://docs.nvidia.com/cuda/cuda-runtime-api/graphs-thread-safety.html#graphs-thread-safety). So we dis-allowed it till now as we only had one graph per-session. But technically, using different cuda graph objects on different threads should be supported but some users have reported issues with that as well - https://stackoverflow.com/questions/61423391/cudagraph-multi-threaded-stream-capturing-causes-errors-only-when-run-in-cuda-m